src/state_manager/mod.rs

```diff
@@ -1122,6 +1126,85 @@ where
         resolve_to_key_addr(&state, self.blockstore(), addr)
     }
 
+    pub async fn miner_get_base_info(
+        self: &Arc<Self>,
+        beacon_schedule: Arc<BeaconSchedule>,
+        tipset: Arc<Tipset>,
+        addr: Address,
+        epoch: ChainEpoch,
+    ) -> anyhow::Result<Option<MiningBaseInfo>> {
+        let prev_beacon = self
+            .chain_store()
+            .chain_index
+            .latest_beacon_entry(&tipset)?;
+
+        let entries: Vec<BeaconEntry> = beacon_schedule
+            .beacon_entries_for_block(
+                self.chain_config.network_version(epoch),
+                epoch,
+                tipset.epoch(),
+                &prev_beacon,
+            )
+            .await?;
+
+        let base = entries.last().unwrap_or(&prev_beacon);
+
+        let (lb_tipset, lb_state_root) = ChainStore::get_lookback_tipset_for_round(
+            self.cs.chain_index.clone(),
+            self.chain_config.clone(),
+            tipset.clone(),
+            epoch,
+        )?;
+
+        let actor = self
+            .get_actor(&addr, *tipset.parent_state())?
+            .ok_or_else(|| Error::State("miner actor does not exist".to_string()))?;
```

Comment:
We can directly use `&str` errors if we aren't returning a `state_manager::Error`.

```suggestion
            .context("miner actor does not exist")?;
```